gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StorageStubProvider.java

```diff
@@ -146,6 +174,9 @@ private ChannelAndRequestCounter buildManagedChannel() {
             .enableRetry()
             .defaultServiceConfig(getGrpcServiceConfig())
             .intercept(counter)
+            // TODO(weiranf): Temporarily added for debugging,
+            //                will be removed once GCS/DirectPath is stable.
+            .intercept(new RemoteAddressDebuggingInterceptor())
```

Comment:
Is the effect of the interceptor here to populate remote address into read/write error status? Make sure you have manually tested it to work.

Reply:
I have tested manually with fake error status. For read, the exception always got caught with right message. For write, I only see errors got logged for non-transient errors. And I noticed that [getResponseOrThrow](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/master/gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java#L369) only throws exception for non-transient errors. For transient errors, the client just retries without displaying any loggings. Is that expected?

Reply:
Yes this is expected, it will eventually throw non retriable error if all retries also fail.